fscollections/forms.py

```diff
@@ -0,0 +1,357 @@
+            self.add_error(field=None, error=forms.ValidationError("You don't have permissions to edit this collection"))
+        else: 
+            if cleaned_data['name'] != self.instance.name:
+                if Collection.objects.filter(user=self.instance.user, name=cleaned_data['name']).exists():
+                    self.add_error('name', forms.ValidationError("You already have a collection with this name"))
+                elif cleaned_data['name'].lower() == 'bookmarks' or cleaned_data['name'].lower() == 'bookmark':
+                    self.add_error('name', forms.ValidationError("This collection name is booked for your personal default collection. Please choose another one."))
+        collection_sounds = self.cleaned_data.get('collection_sounds').split(',')    
+        if len(collection_sounds) > settings.MAX_SOUNDS_PER_COLLECTION:
+            self.add_error('collection_sounds', forms.ValidationError(f'You have exceeded the maximum number of sounds for a collection ({settings.MAX_SOUNDS_PER_COLLECTION}).'))
+            cleaned_data['collection_sounds'] = collection_sounds[:self.instance.num_sounds]
+        return cleaned_data
+
+    def clean_ids_field(self, field):
+        # this function cleans the sounds and maintainers fields which store temporary changes on the edit URL
+        new_field = re.sub("[^0-9,]", "", self.cleaned_data[field])
+        new_field = re.sub(",+", ",", new_field)
+        new_field = re.sub("^,+", "", new_field)
+        new_field = re.sub(",+$", "", new_field)
+        if len(new_field) > 0:
+            new_field = {int(usr) for usr in new_field.split(",")}
+        else:
+            new_field = set()
+        return new_field
+    
+    def save(self, user_adding_sound):
+        """This method is used to apply the temporary changes from the edit URL to the DB.
+        Useful for maintainers and sounds, where several objects are added to the Collection attrs.
+        This way, the server side does not change until the Save Collection button is clicked.
+
+        Args:
+            user_adding_sound (User): the user modifying the collection
+
+        Returns:
+            collection (Collection): the saved collection with proper modficiations
+        """
+        collection = super().save(commit=False)        
+        new_sounds = self.clean_ids_field('collection_sounds')
+        current_sounds = list(Sound.objects.filter(collections=collection).values_list('id', flat=True))
+        for snd in new_sounds:
+            if snd not in current_sounds:
+                sound = Sound.objects.get(id=snd)
+                CollectionSound.objects.create(user=user_adding_sound, sound=sound, collection=collection, status='OK')
+              
+            else:
+                current_sounds.remove(snd)
+        for snd in current_sounds:
+            sound = Sound.objects.get(id=snd)
+            CollectionSound.objects.get(collection=collection, sound=sound).delete()
```

Comment:
this makes 3 queries per item to delete. better to do either:
```py
sounds = Sound.objects.filter(id__in=current_sounds)
collection.sounds.remove(*sounds)
```
(but careful with this option, because it won't trigger a delete signal, because it modifies the m2m field)

or maybe
```py
sounds = Sound.objects.filter(id__in=current_sounds)
CollectionSound.objects.filter(collection=collection, sound__in=sounds).delete()
```
because the filter/delete will do _only_ a delete query and no SELECT.